consul/base.py

```diff
@@ -665,7 +673,12 @@ class Txn(object):
         def __init__(self, agent):
             self.agent = agent
 
-        def put(self, payload):
+        def put(self,
+                payload,
+                token=None,
+                consistency=None,
+                dc=None,
+                decode=False):
```

Comment:
Not sure about this but I prefer to hide encoding/decoding logic from user. This will break API but it will be more clear. Is there any objection against it? Moreover, new parameters should be documented in doc string.

Reply:
fixed